### ISSUE TITLE
loss is nan

### ISSUE THREAD

**Issue by sunnytt98** [huggingface/nanoVLM#165]
When I try to train the model, the loss value becomes NaN, even though I'm using the default parameters. Has anyone else encountered this issue?

`print(f"Batch {i+1}/{len(train_loader)} - Loss: {batch_loss:.4f}")`

```
Batch 706/56191 - Loss: nan
Batch 707/56191 - Loss: nan
Batch 708/56191 - Loss: nan
Batch 709/56191 - Loss: nan
Batch 710/56191 - Loss: nan
Batch 711/56191 - Loss: nan
Batch 712/56191 - Loss: nan
Batch 713/56191 - Loss: nan
Batch 714/56191 - Loss: nan
Batch 715/56191 - Loss: nan
Batch 716/56191 - Loss: nan
Batch 717/56191 - Loss: nan
Batch 718/56191 - Loss: nan
```

**Comment by SUNTAO1963** [comment]
It was found that the issue was caused by the PyTorch version. The loss became NaN when using PyTorch version 2.4.1+cu118. After switching to version 2.6.0 or higher, the loss returned to normal.
